File: src/leetCode/java/Solution190.java
package leetCode.java;

//Bit O(logn) O(1), 因為for迴圈n==0會跳出, 則時間複雜度為以2為底的logn(每次都除以二)
class Solution190 {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int reverse = 0;
        for(int i=0; i < 32 && n != 0; i++){//n為0時直接跳出
            reverse |= ((n & 1) << (31 - i));
            n >>>= 1;
        }
        return reverse;
    }
}

File: src/leetCode/c++/Solution190.cpp
//Bit O(logn) O(1), n每次都除二即logn
class Solution190 {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t reverse = 0;
        for(int i=0; i < 32 && n != 0; ++i){
            reverse |= ((n & 1) << (31 - i));
            n >>= 1;
        }
        return reverse;
    }
};